packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -3978,6 +3981,20 @@ class ExcludeSemantics extends SingleChildRenderObjectWidget {
   }
 }
 
+class _ExcludeSemanticsElement extends SingleChildRenderObjectElement {
+  _ExcludeSemanticsElement(ExcludeSemantics widget) : super(widget);
+
+  @override
+  ExcludeSemantics get widget => super.widget;
+
+  @override
+  void visitChildrenForSemantics(ElementVisitor visitor) {
+    if (widget.excluding)
+      return;
```

Comment:
This PR proposes that we marry the two, which may not be what we want, hence multiple proposals and the discussion. The way we could look at it is "semantics" describes the _meaningful parts_ or _business logic_ of the application. The description of the UI begins with a structural layout (the `Element` tree) then transformed into a visual layout (the `RenderObject` tree). This transformation loses some of the structural information but gains visual information. You can extract the meaningful parts from either of these trees, but you can only extract full structural information from the `Element` tree, and you can only extract full visual information from the `RenderObject` tree. In both cases it's "semantics", but the sets are different, even if the two have a non-empty intersection (e.g. text).

For the purposes of accessibility we use the semantics provided by the render object tree. For the purposes of unit testing element tree semantics probably makes sense in the default case (black box testing), with an option to opt-out (white box testing). For driver tests we probably want the intersection of the two.